mobile/encryption.go

```diff
@@ -0,0 +1,87 @@
+package mobile
+
+import (
+	"io"
+	"crypto/rand"
+	"golang.org/x/crypto/nacl/secretbox"
+)
+
+
+type Encryption struct {
+	EncryptedData []byte
+	DecryptedData []byte
+	ErrorCode int
+
+	// ErrorCode Descriptions
+	// 0 => No Error
+	// 1 => Invalid SecretKey
+	// 2 => Random Generator Error
+	// 3 => Invalid encrypted data length
+	// 4 => Decryption error
+}
+
+func Encrypt(data []byte, sk []byte) *Encryption  {
+	var secretKey [32]byte
+
+	if len(sk) != len(secretKey) {
+		return &Encryption{
+			ErrorCode: 1,
+		}
+	}
+
+	copy(secretKey[:], sk)
```

Comment:
Actually I used `[32]byte` because of secretbox's functions wait fixed size(32) byte array. In order to achieve, I need to copy sk to secretKey.

Reply:
But are you not passing `sk` as a 32 byte array anyway?

Reply:
I'm passing secretKey as a 32 byte array. You can find the function details [here](https://pkg.go.dev/golang.org/x/crypto/nacl/secretbox#Open)

Reply:
I thought you could just do this:

```
encrypted := secretbox.Seal(nonce[:], data, &nonce, &sk)
```
instead of
```
encrypted := secretbox.Seal(nonce[:], data, &nonce, &secretKey)
```

Are you saying it doesn't work because `Seal` requires a fixed length []byte and `sk` is not a fixed length? Can you not pass `sk` to `Encrypt` as a fixed length?

Reply:
Exactly, Seal and Open require fixed length byte array and sk is not fixed and cannot pass sk directly. 

```
cannot use sk (variable of type []byte) as type *[32]byte in argument to secretbox.Seal
```

Reply:
As far as I remember, when I set the argument as `func Decrypt(data []byte, sk [32]byte) *Encryption {`

It will be seen as unsupported parameter that's why we need not fixed length of byte array instead of above.